### PR TITLE
[DataGrid] Improve sorting accessibility

### PR DESCRIPTION
**Pull request by cherniavskii**
Part of #4124

As suggested in the issue, to indicate that columns are sortable `aria-sort` should be always set.
The default value is `none` (as per https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-sort)

### CODE REVIEW

**Review by m4theushw — COMMENTED** [review]
I don't know if we should push this forward or not. The demo in https://www.w3.org/TR/wai-aria-practices-1.1/examples/grid/dataGrids.html#ex2_label also has `aria-sort=none` but NVDA doesn't read the column as "sortable". There're some inconsistencies on how screen readers should read this value: https://github.com/w3c/aria-practices/issues/1192

File: packages/grid/x-data-grid/src/components/columnHeaders/GridColumnHeaderItem.tsx [hunk 1]
```diff
@@ -156,7 +156,7 @@ function GridColumnHeaderItem(props: GridColumnHeaderItemProps) {
 
   const width = column.computedWidth;
 
-  let ariaSort: 'ascending' | 'descending' | undefined;
+  let ariaSort: 'ascending' | 'descending' | 'none' = 'none';
```
Comment (DanailH) [inline packages/grid/x-data-grid/src/components/columnHeaders/GridColumnHeaderItem.tsx:159]: Wouldn't just have it as `none` work?

**Comment by DanailH** [comment]
> I don't know if we should push this forward or not. The demo in https://www.w3.org/TR/wai-aria-practices-1.1/examples/grid/dataGrids.html#ex2_label also has `aria-sort=none` but NVDA doesn't read the column as "sortable". There're some inconsistencies on how screen readers should read this value: [w3c/aria-practices#1192](https://github.com/w3c/aria-practices/issues/1192)

If `aria-sort=none` is a valid state then we can add it I think. Then comes the question if it will actually help the screen reader to read out that a column is sortable.

File: packages/grid/x-data-grid/src/components/columnHeaders/GridColumnHeaderItem.tsx (hunk 1 above)
Comment (cherniavskii) [reply packages/grid/x-data-grid/src/components/columnHeaders/GridColumnHeaderItem.tsx:159]: Not sure what you mean. One `'none'` is for types and another one is used as a value for `ariaSort`

**Comment by cherniavskii** [comment]
Indeed, screen readers just omit sort information when the value is `none`, see https://a11ysupport.io/tech/aria/aria-sort_attribute

Screenshot 2022-04-28 at 11 42 31

But at least we do what we can - I don't see any drawbacks in supporting `none` value.

**Review by DanailH — APPROVED** [review]
I'm ok with merging this. It won't solve the issue but it is a step.

**Pull request merged**